processing/src/main/java/org/apache/druid/segment/DimensionDictionary.java

```diff
@@ -160,4 +173,16 @@ public int getIdForNull()
       lock.readLock().unlock();
     }
   }
+
+  private long getObjectSize(@Nonnull T object)
```

Comment:
Thanks for pointing this out! I will see how we can make this cleaner.

Reply:
Used the `StringDimensionDictionary` suggestion, although I have not made it abstract so that implementations using the `DimensionDictionary` can continue to use it as the base concrete class.